src/include/simeng/PerceptronPredictor.hh

```diff
@@ -43,11 +44,27 @@ class PerceptronPredictor : public BranchPredictor {
   void flush(uint64_t address) override;
 
  private:
-  /** The bitlength of the BTB index; BTB will have 2^bits entries. */
+  /** Returns the dot product of a perceptron and a history vector.  Used to determine
+   * a direction prediction */
+  int64_t getDotProduct(std::vector<int8_t> perceptron, uint64_t history) {
```

Comment:
its probably worth passing the `perceptron` vector as a const reference given we only read from it. It won't effect performance too much either way,  but still worth changing in my opinion

Also, for codebase consistency, could you only have the function declaration in the header file and then define the function in the `.cc`